File: search_swagger.js

const fs = require('fs');
try {
    const data = JSON.parse(fs.readFileSync('c:/Users/user/Desktop/imedteam/imedteamuz/swagger.json', 'utf8'));

    const paths = data.paths || {};
    console.log("Matching Paths:");
    Object.keys(paths).forEach(path => {
        if (path.toLowerCase().includes('promo') || path.toLowerCase().includes('code') || path.toLowerCase().includes('voucher') || path.toLowerCase().includes('coupon')) {
            console.log(path);
            Object.keys(paths[path]).forEach(method => {
                console.log(`  ${method}: ${paths[path][method].summary || ''}`);
            });
        }
    });

    // also check /mobile/order structure
    if (paths['/mobile/order']) {
        console.log('\n/mobile/order methods:', Object.keys(paths['/mobile/order']));
    }

} catch (e) {
    console.error(e);
}

File: search_swagger.py

import json
import os

try:
    with open('c:/Users/user/Desktop/imedteam/imedteamuz/swagger.json', 'r', encoding='utf-8') as f:
        data = json.load(f)

    paths = data.get('paths', {})
    definitions = data.get('definitions', {})

    print("--- Matching Paths ---")
    for path in paths:
        if any(x in path.lower() for x in ['promo', 'coupon', 'discount', 'code', 'order']):
            print(f"Path: {path}")
            for method in paths[path]:
                print(f"  Method: {method}")
                # print(json.dumps(paths[path][method], indent=2))

    print("\n--- Matching Definitions ---")
    for def_name in definitions:
        if any(x in def_name.lower() for x in ['promo', 'coupon', 'discount', 'code', 'order']):
            print(f"Definition: {def_name}")
            # print(json.dumps(definitions[def_name], indent=2))

except Exception as e:
    print(f"Error: {e}")
